### PR TITLE
Fix inconsistent Intel hardware intrinsic APIs

### PR DESCRIPTION
**Pull request by fiigii**
- Fixes https://github.com/dotnet/coreclr/issues/18831 
- Correct SSE2/AVX2 `SumAbsoluteDifferences` return type from `Vector128<long/ulong>` to `Vector128<ushort>`.
- Rename SSE2 `MultiplyHorizontalAdd` to `MultiplyAddAdjacent` to keep consistent with SSSE3/AVX2 counterparts.

### CODE REVIEW

**Comment by fiigii** [comment]
@CarolEidt @tannergooding @eerhardt

**Comment by fiigii** [comment]
If you guys think the API change is okay, I would disable their tests in this PR as well.

File: src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Sse2.cs [hunk 1]
```diff
@@ -1313,7 +1313,7 @@ public static unsafe Vector128<double> SetVector128(double e1, double e0)
         /// __m128i _mm_sad_epu8 (__m128i a,  __m128i b)
         ///   PSADBW xmm, xmm/m128
         /// </summary>
-        public static Vector128<long> SumAbsoluteDifferences(Vector128<byte> left, Vector128<byte> right) => SumAbsoluteDifferences(left, right);
```
Comment (tannergooding) [inline src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Sse2.cs]: Shouldn't this be `Vector128<ushort>`?

> Computes the absolute differences of the
packed unsigned byte integers from xmm2
/m128 and xmm1; the 8 low differences and 8
high differences are then summed separately
to produce two unsigned word integer results.

Comment (fiigii) [reply src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Sse2.cs]: Ah, right! Will fix. 
Thank you!

Comment (fiigii) [reply src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Sse2.cs]: Done. Now they are correct and consistent with `MultipleSumAbsoluteDifferences`.

**Comment by fiigii** [comment]
Yes, I am looking for more to make sure we fix all the inconsistency.

**Comment by BruceForstall** [comment]
@dotnet/dnceng Failure in https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_checked_ubuntu_innerloop_flow_prtest/7594/ is "java.io.IOException: No space left on device"

**Comment by fiigii** [comment]
I checked that this kind of inconsistency is all fixed in this PR. 
CI failure not related.

Can we merge?

**Pull request merged**